### PR TITLE
RSDK-5318 - Update proto conversions with new module fields

### PR DESCRIPTION
**Pull request by zaporter-work**
Ticket: https://viam.atlassian.net/browse/RSDK-5318

This brings the new module fields to the rdk.

This depends on https://github.com/viamrobotics/goutils/pull/209

I have explicitly only updated the proto conversions because `app` uses the rdk proto conversions so as long as this makes it into the upcoming release, development should not be blocked.

### CODE REVIEW

File: module/modmanager/manager.go
Comment (zaporter-work) [inline module/modmanager/manager.go:1]: These changes are not  required and if anyone had any qualms, I can revert everything changed in this file. I just need the proto changes to get merged

Comment (michaellee1019) [reply module/modmanager/manager.go:1]: Its fine IMO. It was more concerning about the API protos having fields we didn't have exact plans for.

**Review by michaellee1019 — APPROVED** [review]

File: config/module.go [hunk 1]
```diff
@@ -27,11 +27,28 @@ type Module struct {
 	// value besides "" or "debug" is used for LogLevel ("log_level" in JSON). In other words, setting a LogLevel
 	// of something like "info" will ignore the debug setting on the server.
 	LogLevel string `json:"log_level"`
+	// Type indicates whether this is a local or registry module.
+	Type ModuleType `json:"type"`
+	// ModuleID is the id of the module in the registry. It is empty on non-registry modules.
+	ModuleID string `json:"module_id,omitempty"`
+	// Environment contains additional variables that are passed to the module process when it is started.
+	// They overwrite existing environment variables.
+	Environment map[string]string `json:"env,omitempty"`
```
[the remark this answers is not in the document]
Comment (zaporter-work) [reply config/module.go:36]: Are you talking about the `command.Env` field and the fact that it is `[]string`? 

The golang os package expects us to pass arguments to a command in a []string that is `"KEY_NAME=keyvalue"`. However that isn't terribly ergonomic (doing things like listing all the  env vars keys would require string parsing) so I've tried to isolate that to an implementation detail inside of goutils and keep just a simple map[string]string until the last moment.

**Review by RoxyFarhad — APPROVED** [review]
lgtm! just had a quick about string <> map

File: config/module.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (RoxyFarhad) [reply config/module.go:36]: another potentially dumb q: but is the goutils where we define the processConfig? I assumed that was all done in the RDK tbh

[the remark this answers is not in the document]
Comment (zaporter-work) [reply config/module.go:36]: process config is defined in `goutils/pexec/process.go` and then started in `goutils/pexec/managed_process.go`. I agree that is is a surprising cross-repo dependency

**Pull request merged**